cmake/modules/Findnl.cmake

```diff
@@ -0,0 +1,56 @@
+# - Find libnl
+# Find the libnl-3 library and includes
+#
+# nl_INCLUDE_DIR - where to find netlink.h, etc.
+# nl_<COMPONENT>_LIBRARY - library when using nl::<COMPONENT>.
+# nl_FOUND - True if nl found.
+
+find_path(nl_INCLUDE_DIR
+  NAMES
+    netlink/netlink.h
+  PATH_SUFFIXES
+    libnl3)
+
+# find the core library by default
+list(FIND nl_FIND_COMPONENTS "core" found)
+if(found EQUAL -1)
+  list(APPEND nl_FIND_COMPONENTS "core")
+endif()
+
+foreach(component ${nl_FIND_COMPONENTS})
+  set(nl_COMPONENTS core cli genl idiag nf route xfrm)
```

Comment:
Hey, I am not very familiar with cmake. That's why the original code looked very basic like the udev/blkid cmake stuff :)

For rbd-nbd we only use genl. Is it necessary to search for these other components or is more a cmake standard type of style?

Reply:
I meant to say we only use libnl and libnl-genl.

Reply:
> Is it necessary to search for these other components

no, it's not necessary. they are listed here only for the sake of completeness.

> we only use libnl and libnl-genl.

yeah, i agree. i would be happy to remove the others if you think it makes the script easier to understand or better.

Reply:
It does not matter to me.